### PR TITLE
update lesson examples

### PR DESCRIPTION
**Pull request by linda-seiter**

### CODE REVIEW

**Review by developerjaym — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -2,31 +2,351 @@
...
-- Learning Goal 1
-- Learning Goal 2
+- Demonstrate runtime polymorphism based on method overriding
+- Demonstrate compile-time polymorphism based on method overloading
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+Polymorphism means "many forms". In Java,
+polymorphism means the ability of an object or method to take many forms.
```
Comment (kcleland0818) [inline README.md]: Consider rewording this sentence to:

"In Java, polymorphism is the ability for an object or method to take on many forms."

File: README.md [hunk 2]
```diff
@@ -2,31 +2,351 @@
...
+    @Override
+    public void swim() {
+        System.out.println("Fish swims using fins");
+    }
+}
+```
+
+```java
+public class Airplane implements Flyable {
+    private String manufacturer;
```
Comment (kcleland0818) [inline README.md:60]: Add an extra new-line here maybe to be consistent with the other three classes?

File: README.md [hunk 3]
```diff
@@ -2,31 +2,351 @@
...
+
+
+- The polymorphic variable `swimmer`  initially references a `Fish` and then is reassigned to reference a `Duck` object.
+- The polymorphic variable `flyer` initially references an  `Airplane` and then is reassigned to reference a `Duck` object.
+
+Runtime polymorphism means the behavior of a method invocation
+can vary depending on the actual type of the object referenced.
+Java uses **dynamic binding** for the method calls `swimmer.swim()`
+and `flyer.fly()`, determining which method `swim()` and `fly()` method
+to execute based on the type of the object referenced by `swimmer` and `flyer`.
```
Comment (kcleland0818) [inline README.md]: "Java uses **dynamic binding** for the method calls `swimmer.swim()` and `flyer.fly()`, determining which method `swim()` and `fly()` method to execute based on the type of the object referenced by `swimmer` and `flyer`."

Consider rewording this sentence to the following:

Java uses **dynamic binding** for the method calls `swimmer.swim()` and `flyer.fly()`. Dynamic binding will determine which `swim()` and `fly()` method to execute based on the type of object referenced by `swimmer` and `flyer`.

File: README.md [hunk 4]
```diff
@@ -2,31 +2,351 @@
...
+
+![polymorphism step4](https://curriculum-content.s3.amazonaws.com/6677/pillars/poly_step4.png)
+
+Continue stepping through the code.  Confirm the two method calls `flyable.fly()`
+invoke different method implementations based on whether `flyable` references
+an `Airplane` or a `Duck` object.
+
+## Runtime Polymorphism and Inheritance
+
+We also encounter polymorphic variables references
```
Comment (kcleland0818) [inline README.md]: Should "variables" be singular?

File: README.md [hunk 5]
```diff
@@ -2,31 +2,351 @@
...
+    public String getWeekendPlans() {
+        return "grade homework assignments";
+    }
+
+}
+```
+
+```java
+public class Student extends Person {
+    private String favoriteSubject;
```
Comment (kcleland0818) [inline README.md:233]: Add an extra new line here to be consistent with the formatting from the `Person` and `Teacher` class?

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md]: fixed.

File: README.md (hunk 2 above)
Comment (linda-seiter) [reply README.md:60]: fixed

File: README.md (hunk 3 above)
Comment (linda-seiter) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (linda-seiter) [reply README.md]: fixed

File: README.md (hunk 5 above)
Comment (linda-seiter) [reply README.md:233]: fixed

**Review by kcleland0818 — APPROVED** [review]

**Review by kcleland0818 — APPROVED** [review]

**Review by AlveeM — APPROVED** [review]

**Pull request merged**